### ISSUE TITLE
Dutch translation

### ISSUE THREAD

**Issue by hoi3** [axelboc/anki-ultimate-geography#313]
Thanks for the amazing geography deck! 
I am considering making a Dutch translation of the deck. However, I am not at all familiar with GitHub, Anki Deck Manager and other aspects that are involved in the process. I've read the CONTRIBUTING.md but still I'm not sure how to proceed exactly. I have installed the CrowdAnki addon and Anki Deck Manager but I'm a bit lost from there on. I If you could explain a bit more detailed how I should approach this project, I think I'd be able to make my translation publicly available. I am planning to make Dutch translations anyway, so I think it would be worth it for me to put a little extra effort into this so that other people can use the translated deck as well.

**Comment by axelboc** [comment]
That's very nice of you! 💯 

I'd love to tell you how to get started but I'm afraid adding a sixth translation may not be the best idea at this point in time... 😓

Firstly, we're currently in the process of removing a bunch of content from the deck - cf. #312. So it'd be a shame for you to translate something that'll end up being removed.

Secondly, because of the way the deck is currently set up, with Anki Deck Manager, all translations are managed within a single CSV file: `src/data.csv`. That's five languages cramped into a single CSV file... I won't get into the details, but in short, we've reached the limit of our set up... Just open it and you'll understand. 😄 

But hope is in sight! One of our contributors, @ohare93, is working on a new deck management system called _Brain Brew_, which should make it much easier to manage translations. If you'd like to know more, check out issue #143. I can't give you an estimate as to when we'll migrate to Brain Brew or when we'll be ready to integrate new translations -- all I can say is that it may very well take a while.

---

Now, it'd be a shame for you to translate the deck anyway, as you seem to want to do, and not be able to share it!

The way I see it, if you could get your translations into a CSV file similar to `data.csv` and upload it to this issue, then I'm sure we'd manage to integrate them into the deck once it's ready to receive more translations. Sure, some of the translations would become outdated, but I reckon it wouldn't be too much effort to go through and update them when the time comes.

How to get your translations into a CSV file is really up to you. To get started, you'd probably want to proceed as follows:

1. First wait for PR #312 to be merged, or better, if you're patient, for v4.0 of the deck to be released, which should happen within a month.
1. Make a copy of the deck's `src/data.csv` file.
1. Make some space in the file by deleting the columns that belong to other translations (`Country:de`, `Country:es`, etc.).
1. Add the appropriate columns for your translation: `Country:nl`, `"Country info:nl", etc.
1. Double check the contributing guidelines to make sure you understand how to choose country names and capitals on Wikipedia, notably.
1. Start translating!

The last step can be done manually, of course, but there are probably ways to automate it if you've got the skills. One of our contributors, @ukanuk, has notably been working on a script to fetch country names and capitals from Wikipedia by following interwiki links. It's not full proof, but it could give you a head start.

I hope this gives you a way forward. Let us know if you have any questions or if any of this wasn't clear. Most importantly, thanks again and good luck to you!! 💪 🚀

**Comment by hoi3** [comment]
Thanks for the reply. I'll wait at least for V4.0 then before I start. When that's released I'll check back what the best approach would be. I might start translating then slowly at my own pace as (if I understand correctly) it might be wisest to wait with releasing the translation until managing the translations becomes easier with _Brain Brew_.

**Comment by WouterVdS** [comment]
Hi there! As PR #312 is merged, it seems like now is a good time to add Dutch? I would like to contribute and will start working if you guys confirm the timing is right!

**Comment by axelboc** [comment]
You can indeed start the translation by following the steps in my previous comment. Once the deck is migrated to Brain Brew, we'll integrate the translation and release it. Good luck!

**Comment by ohare93** [comment]
If this version is going to be merged into Brain Brew first thing anyways, perhaps it is easier for you @WouterVdS to just make the translation in Anki 👍 rather than you playing around with a messy gigantic Csv that will soon be outdated. We can import your new Dutch data into the new Csvs ourselves, with no issue 👍 this is an option though, feel free to use the old way if you prefer!

What do you think @axelboc? 🙂

**Comment by WouterVdS** [comment]
I think working with the CSV is a little bit faster for me!

**Comment by WouterVdS** [comment]
I found time to create the Dutch translation! But I see the Brain Brew pr is not yet merged. Do I wait on this merge?

**Comment by aplaice** [comment]
I think that it'd be best to wait on the merge.

<hr/>

The "`anki-dm` CSV" format (our current one) is easily interconvertible with the "`Brain Brew` CSV" format, so you could in principle open a PR with the Dutch translation now and later convert it to the Brain Brew format, but that would be a bit of wasted/unnecessary effort and extra friction, so I don't think it's worth it.

Hopefully, we'll be merging the Brain Brew PR soon!

(Thanks for still being eager!)

**Comment by ohare93** [comment]
@WouterVdS New deck manager is pulled :+1: The source csv has now been broken into multiple separate csv files inside `src/data`. 

You can now add your translation :clap: This requires 2 steps to do it yourself:
1) Add the translation as a column in each of the source files.
2) Edit the Brain Brew recipe in `recipes/source_to_anki.yaml` to build your new column to your new language deck. Just copy pasting what the others do, and changing it to match your new columns will do :+1: 

I am happy to do the second step, if you do the first :wink: 

Better yet, if you have created the translation inside Anki itself then I can do it all in <5mins, as the new deck manager can also import _from CrowdAnki_ to our Source control. Feel free to export the deck, upload a zip of it here, and I will do magic :clap: :grin:

**Comment by WouterVdS** [comment]
See pull request #378
